testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replicator_encryption_hook.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import time
+
+from keywords.MobileRestClient import MobileRestClient
+# from keywords.ClusterKeywords import ClusterKeywords
+# from keywords import couchbaseserver
+# from keywords.utils import log_info, random_string, get_embedded_asset_file_path
+# from CBLClient.Database import Database
+from CBLClient.Replication import Replication
+from CBLClient.Document import Document
+# from CBLClient.Authenticator import Authenticator
+# from concurrent.futures import ThreadPoolExecutor
+# from CBLClient.Blob import Blob
+# from CBLClient.Dictionary import Dictionary
+# from libraries.testkit.prometheus import verify_stat_on_prometheus
+# from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from keywords import document
+from libraries.testkit import cluster
+# from utilities.cluster_config_utils import persist_cluster_config_environment_prop, copy_to_temp_conf
+# from keywords.attachment import generate_2_png_100_100
+# from keywords.SyncGateway import SyncGateway
+
+
+@pytest.mark.listener
+@pytest.mark.parametrize("type", [
+    ("push")
+])
+def test_replication_with_encryption(params_from_base_test_setup, type):
+    """
+    @summary:
+    1. Have SG and CBL up and running
+    2. Create a simple document with encryption property
+    3. Start the replicator and make sure documents are
+    replicated on SG
+
+    7. Update the normal data
+    8. Start the replicator and verify the docs on CBL
+
+    10.Delete the document on CBL and replicate
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    num_of_docs = 1
+    number_of_updates = 2
+    sg_client = MobileRestClient()
+
+    if sync_gateway_version < "2.0.0":
+        pytest.skip('This test cannot run with sg version below 2.0')
+    channels_sg = ["ABC"]
+    username = "autotest"
+    password = "password"
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # Create docs in CBL
+    channel = ["Replication-1"]
+    db.create_bulk_docs(num_of_docs, "cbl", db=cbl_db, channels=channels_sg)
+    doc_id = "doc_1"
+    documentObj = Document(base_url)
+    doc_body = document.create_doc(doc_id=doc_id, content="doc1", channels=channel, cbl=True)
+    doc1 = documentObj.create(doc_id, doc_body)
+    db.saveDocument(cbl_db, doc1)
+
```

Comment:
I didn't see any encryption has been specified in this test case, what this test is for???